### ISSUE TITLE
Implement upstream

### ISSUE THREAD

**Issue by andia89** [skjerns/mat7.3#33]
Great package, solved a lot of problems for me. SInce the matlab version 7.3 has been around a while now, did you consider implementing your changes upstream to scipy directly?

**Comment by skjerns** [comment]
Yes, this was discussed in https://github.com/scipy/scipy/issues/11351

Main points:
* would induce optional dependency on h5py
* would have to be backwards-compatible, which would need a lot of work
   * currently scipy.io.loadmat returns quite silly data types. I think it's solved better in `mat73`, however I would need to keep with the previous functionality

so, in summary, nope, it's probably not going to happen soon :-/

**Comment by andia89** [comment]
Ah I see.

At least it seems that your main concern about silly return data types has been solved, no?
https://blog.nephics.com/2019/08/28/better-loadmat-for-scipy/ (the changes proposed here seem to have been implemented)

I looked a bit at the scipy code and indeed it would seem it is quite an undertaking to implement it, I just find it weird that the situation has now been like that for quite some time and the scipy devs don't seem to think it is a priority

**Comment by skjerns** [comment]
nice to see that there is progress w.t.r! 

However, I think it's still quite cumbersome having to program all the backwards compatibility and kwarg combinations that the mess that is `scipy.io.loadmat` brings with it. 

* Maybe one day I'll get to it o
* If the Scipy team thinks that it would be good to integrate it somehow, I'd also be up for it.
